cli/src/main/scala/com/advancedtelematic/tuf/cli/repo/TufRepo.scala

```diff
@@ -131,6 +165,64 @@ class TufRepo(val name: RepoName, val repoPath: Path)(implicit ec: ExecutionCont
     }
   }
 
+  private def fetchRootChain(reposerverClient: UserReposerverClient, from: Int, to: Int): Future[Vector[SignedPayload[RootRole]]] = {
+    val versionsToFetch = from until to
+
+    val rootRolesF = versionsToFetch.foldLeft(Future.successful(Vector.empty[SignedPayload[RootRole]])) { (accF, vv) =>
+      for {
+        acc <- accF
+        payload <- reposerverClient.root(vv.some).recoverWith {
+          case RoleNotFound(msg) => Future.failed(RootPullError(NonEmptyList.of(msg)))
+        }
+      } yield acc :+ payload
+    }
+
+    rootRolesF
+  }
+
+  private def validatePath(reposerverClient: UserReposerverClient, from: SignedPayload[RootRole], to: SignedPayload[RootRole]): Future[Unit] = {
+    val rootRolesF = fetchRootChain(reposerverClient, from.signed.version + 1, to.signed.version)
```

Comment:
I might be reading this wrong, but it looks like the validation only happens on the fetch root roles. So if the client has `v1`, and we fetch `v2` to `v5`, then this will only check the pairs `(2, 3)`, `(3, 4)` and `(4, 5)`.

Reply:
Ok, that makes sense. I think I just got a little lost in the scala-iness of it.